src/main/java/org/ow2/proactive/catalog/repository/entity/KeyValueMetadataEntity.java

```diff
@@ -62,25 +62,26 @@
 @Inheritance(strategy = InheritanceType.SINGLE_TABLE)
 public class KeyValueMetadataEntity implements Serializable {
 
-    @GeneratedValue(generator = "uuid2")
-    @GenericGenerator(name = "uuid2", strategy = "uuid2")
-    @Column(name = "ID", columnDefinition = "BINARY(16)")
-    @Id
-    private UUID id;
-
     @Column(name = "KEY", nullable = false)
     protected String key;
 
     @Column(name = "VALUE", nullable = false)
     protected String value;
 
+    @SuppressWarnings("DefaultAnnotationParam")
     @Column(name = "LABEL", nullable = true)
     protected String label;
 
     @ManyToOne
     @JoinColumns({ @JoinColumn(name = "CATALOGOBJECTREVISION", referencedColumnName = "ID") })
     protected CatalogObjectRevisionEntity catalogObjectRevision;
 
+    @GeneratedValue(generator = "uuid2")
+    @GenericGenerator(name = "uuid2", strategy = "uuid2")
+    @Column(name = "ID", columnDefinition = "BINARY(16)")
+    @Id
+    private UUID id;
+
     public KeyValueMetadataEntity(String key, String value, String label) {
```

Comment:
We use the label parameter, when we store these additional information (KeyValueMetadataEntity) inside the database.

I think that parameter is intended to allow the search by generic information or other job related information without reading the xml inside the database. 

Currently we use the key as an uniqueness constraint, which creates problems see this issue: https://github.com/ow2-proactive/catalog/issues/60 . In order to fix that, we could widen the uniqueness constraint to key and label. Which would allow us to have a generic information with the key "name" (because the workflow name is stored inside a KeyValueMetadataEntity with the key = "name" -> that violates the uniqueness constraint). 

But it is probably better to ask people who initially worked on the catalog and made those decisions.@yinan-liu can help us out on this question?